cmake/modules/platform/toolcfg/gnu.cmake

```diff
@@ -161,14 +170,20 @@ function(_omr_toolchain_process_exports TARGET_NAME)
 
 	set(exp_file "$<TARGET_PROPERTY:${TARGET_NAME},BINARY_DIR>/${TARGET_NAME}.exp")
 
-	omr_process_template(
-		"${omr_SOURCE_DIR}/cmake/modules/platform/toolcfg/gnu_exports.exp.in"
-		"${exp_file}"
-	)
-
-if(NOT CMAKE_C_COMPILER_IS_OPENXL)
-	target_link_libraries(${TARGET_NAME}
-		PRIVATE
-			"-Wl,--version-script,${exp_file}")
-endif()
+	# Currently, OpenXL on AIX uses the GNU configuration. In the future, it should
+	# be switched to the OpenXL-specific configuration (as has already been done with
+	# z/OS), but until then, we need to add this special case for OpenXL.
+	if (CMAKE_C_COMPILER_IS_OPENXL)
+		omr_process_template(
+			"${omr_SOURCE_DIR}/cmake/modules/platform/toolcfg/xlc_exports.exp.in"
+			"${exp_file}"
+		)
+		set_property(TARGET ${TARGET_NAME} APPEND_STRING PROPERTY LINK_FLAGS " -Wl,-bE:${TARGET_NAME}.exp")
```

Comment:
This should use `exp_file` like line 187:
```
		set_property(TARGET ${TARGET_NAME} APPEND_STRING PROPERTY LINK_FLAGS " -Wl,-bE:${exp_file}")
```

Reply:
I tried that initially, but upon actually trying to build, I found that the value of `${exp_file}` at this point is  `$<TARGET_PROPERTY:jvmtitest,BINARY_DIR>/jvmtitest.exp`. I'm not sure if that's the intended behavior, but that's why I used `${TARGET_NAME}.exp` here instead.

Reply:
The same thing should happen to that string as does for the one below - set a target property that is processed later during makefile generation. You need to look at that generated makefile to see the final result.

Reply:
This can use `${TARGET_NAME}.exp` until we understand why the generator expression does not work